### PR TITLE
vimPlugins.himalaya-vim: fix failing build

### PR DESCRIPTION
**Pull request by toastal**
I’m unsure as to why the configure phase `cd vim/` is no longer working, but this patch phase copying files up a directory workaround is building for my NixOS locally. Configure phase and build phase seemed to be being skipped (??); I couldn't get even an `echo` out of these.

###### Motivation for this change

Himalaya build was failing to build.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
See https://github.com/NixOS/nixpkgs/blob/master/pkgs/misc/vim-plugins/build-vim-plugin.nix#L56-L59

File: pkgs/misc/vim-plugins/overrides.nix [hunk 1]
```diff
@@ -66,7 +66,7 @@
 , openssl
 , pkg-config
 
-# vim-go dependencies
+  # vim-go dependencies
```
Comment (SuperSandro2000) [inline pkgs/misc/vim-plugins/overrides.nix:69]:
```suggestion
# vim-go dependencies
```

File: pkgs/misc/vim-plugins/overrides.nix [hunk 2]
```diff
@@ -51,7 +51,7 @@
 , CoreFoundation
 , CoreServices
 
-# nvim-treesitter dependencies
+  # nvim-treesitter dependencies
```
Comment (SuperSandro2000) [inline pkgs/misc/vim-plugins/overrides.nix:54]:
```suggestion
# nvim-treesitter dependencies
```

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 132517` at 45072144 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>spacevim</li>
    <li>tests.vim.test_vim_with_vim_nix</li>
    <li>vimPlugins.himalaya-vim</li>
  </ul>
</details>
<details>
  <summary>4 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/misc/vim-plugins/build-vim-plugin.nix:28:14:
```
   |
28 |       inherit unpackPhase configurePhase buildPhase addonInfo preInstall postInstall;
   |              ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/license-missing.md">license-missing</a>

Package is missing a license.

Near pkgs/misc/vim-plugins/build-vim-plugin.nix:28:14:
```
   |
28 |       inherit unpackPhase configurePhase buildPhase addonInfo preInstall postInstall;
   |              ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`configurePhase` should probably contain `runHook preConfigure` and `runHook postConfigure`.

Near pkgs/misc/vim-plugins/build-vim-plugin.nix:28:14:
```
   |
28 |       inherit unpackPhase configurePhase buildPhase addonInfo preInstall postInstall;
   |              ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/misc/vim-plugins/build-vim-plugin.nix:28:14:
```
   |
28 |       inherit unpackPhase configurePhase buildPhase addonInfo preInstall postInstall;
   |              ^
```
</li>
  </ul>
</details>

**Pull request merged**